dbt/include/redshift/macros/adapters.sql

```diff
@@ -51,6 +53,22 @@
   as (
     {{ sql }}
   );
+  {% else%}
```

Comment:
Can you add `begin` and `commit` syntax to make this an official database transaction that will automatically rollback if any portion fails? This is similar to what I do in the postgres example above. 

Write a passing test to verify redshift automatically rollbacks to a previous good state for a table. 

https://docs.aws.amazon.com/redshift/latest/dg/r_BEGIN.html
https://docs.aws.amazon.com/redshift/latest/dg/r_COMMIT.html

Reply:
added these explicitly -- don't we implicitly have these as part of the materialization itself though? I think both postgres and redshift effectively operate with this flow

```
begin;
create intermediate;
rename existing to backup;
rename intermediate to existing/target;
commit;
```

is that sufficient, or is the second begin/commit still required?

Reply:
Can you link the file where this code resides? If what you say is true, I may be nesting a transaction within a transaction that may not be needed! Good catch!

Reply:
it's just the [standard table materialization](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/include/global_project/macros/materializations/models/table/table.sql)!

[begin](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/include/global_project/macros/materializations/models/table/table.sql#L26);
[commit](https://github.com/dbt-labs/dbt-core/blob/main/core/dbt/include/global_project/macros/materializations/models/table/table.sql#L51);

Reply:
I removed this in postgres and verified in a couple dry runs that I don't need the `begin` and `commit` syntax! I'll write a formal passing test, but you can test it with a couple dry runs for forced errors to see if it leaves the previous table in place. If yes, you can remove the `begin` and `commit` syntax!